src/main/java/org/javamodularity/moduleplugin/tasks/CompileTask.java

```diff
@@ -29,6 +30,11 @@ private void configureCompileJava(JavaCompile compileJava) {
             } else {
                 configureModularityForCompileJava(compileJava, moduleOptions);
             }
+            helper().findTask(CompileJavaTaskMutator.COMPILE_GROOVY_TASK_NAME, GroovyCompile.class)
+                    .ifPresent(compileGroovy -> {
+                        compileGroovy.getDependsOn().remove(JavaPlugin.COMPILE_JAVA_TASK_NAME);
```

Comment:
Again, would you please explain why it's handled differently for Kotlin and differently for Groovy?

Reply:
The Kotlin plugin configures:
[`compileJava.dependsOn(compileKotlin)`](https://github.com/JetBrains/kotlin/blob/15feeddc4618375ded48bb776c31843d6758b32a/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt#L1001)
while the Groovy plugin configures:
[`compileGroovy.dependsOn(compileJava)`](https://github.com/gradle/gradle/blob/da4afb6813b10c5fb60a84862adff747323dcab6/subprojects/plugins/src/main/java/org/gradle/api/plugins/GroovyBasePlugin.java#L117)

Without changing this dependency the module-info.java would be compiled before compiling the Groovy classes and it will fail.

Reply:
I see, thanks!